src/app/metadata/features/metadata-browser/metadata-browser.component.ts

```diff
@@ -65,27 +71,59 @@ export class MetadataBrowserComponent implements OnInit {
    * On init, define the default values of the search variables
    */
   ngOnInit(): void {
-    this.#updateMetadataServiceSearchTerms();
+    const { pageSize, s, f, skip } = this.#router.routerState.snapshot.root.queryParams;
+    if (pageSize) {
+      this.pageSize = parseInt(pageSize);
+    } else {
+      this.pageSize = DEFAULT_PAGE_SIZE;
+    }
+    if (s) {
+      this.searchTerm = s;
+    }
+    if (f) {
+      const paramVals = this.#facetDataFromString(decodeURIComponent(f));
+      console.log(paramVals);
+      if (paramVals) {
+        this.facetData = paramVals;
+      }
+    }
+    if (skip) {
+      this.#skip = parseInt(skip);
+    }
+
+    this.#updateMetadataServiceSearchTermsAndRoute();
   }
 
   /**
    * Syncs the data between this component and the search service and initiates a new search.
    */
   #performSearch(): void {
-    this.#updateMetadataServiceSearchTerms();
+    this.#updateMetadataServiceSearchTermsAndRoute();
   }
 
   /**
    * Pushes the local filters, search term and page setup to the search service.
    */
-  #updateMetadataServiceSearchTerms(): void {
+  #updateMetadataServiceSearchTermsAndRoute(): void {
     this.#metadataSearch.loadQueryParameters(
       this.#className,
       this.pageSize,
       this.#skip,
       this.searchTerm,
       this.facetData,
     );
+    this.#router.navigate([], {
+      relativeTo: this.#route,
+      queryParams: {
+        skip: this.#skip !== DEFAULT_SKIP_VALUE ? this.#skip : undefined,
+        q: this.searchTerm !== '' ? this.searchTerm : undefined,
+        f:
+          Object.keys(this.facetData).length !== 0
+            ? encodeURIComponent(this.#facetDataToString(this.facetData))
+            : undefined,
+        pageSize: this.pageSize !== DEFAULT_PAGE_SIZE ? this.pageSize : undefined,
+      },
+    });
```

Comment:
See above. Would be more elegant to only change the route and then use the same method as on init to load the new query params.